### ISSUE TITLE
Invalid variable name when bundling with minify=true

### ISSUE THREAD

**Issue by janicduplessis** [facebook/metro#201]
When bundling with minify=true there is an issue with a module in react-native-svg that uses a variable named `r`.
(https://github.com/react-native-community/react-native-svg/blob/master/lib/Matrix2D.js#L230).

Seems to be caused by 215e037c830852f8e965b1ad5ea6ff244323e2e0 or 4839038bb63c1dc5eb9f1c46acc3494af79a77c2.

Without minify
```js
if (rotation % 360) {
  var r = rotation * DEG_TO_RAD;
  cos = Math.cos(r);
  sin = Math.sin(r);
}
```

With https://github.com/facebook/metro/blob/master/packages/metro/src/JSTransformer/worker.js#L279 branch forced (added `true ||`)
```js

if (rotation % 360) {
  var _r13 = rotation * DEG_TO_RAD;

  cos = Math.cos(_r12); // <- _r12 does not exist
  sin = Math.sin(_r12);
} 

```

cc @mjesun @rafeca

**Comment by mjesun** [comment]
Hi!

That was fixed in one of our latest paths. It is due to a constant renaming for `require` as `r` in order to optimize for GZip compression (having a constant prelude showed a somewhat significant improvement).

Which Metro version are you using?

**Comment by janicduplessis** [comment]
@mjesun 0.42.1 which is the latest release.

I managed to get a smaller repro.

```js
import React from 'react';
import { AppRegistry } from 'react-native';

class Test {
  test() {
    if (true) {
      const r = 1;
      return r;
    }
  }
}

const a = new Test();
a.test();

class App extends React.Component {
  render() {
    return null;
  }
}

AppRegistry.registerComponent('app', () => App);
```

Seems like a weird edge case, maybe a bug in babel? It has to have a class and an if statement or the bug doesn't trigger.

**Comment by mjesun** [comment]
Gotcha. I mixed this with an issue we got a few days ago where `require` was wrongly renamed to `r`. I will try to repro on a local instance and see if I can understand what's going on. TBH it looks like a bug in Babel, but let's double-check.

**Comment by mjesun** [comment]
Ok, I saw what's going on. The `constant-folding` plugin is removing the `if (true)`, but the scope is preserved through the block (we keep the `{`, `}`). When whe re-crawl with a `Scope` visitor, we're visiting them, and somehow the renaming logic gets confused and does invalid renaming.

The fix is to perform a `path.scope.crawl()` on the [`Scope` visitor](https://github.com/facebook/metro/blob/master/packages/metro/src/JSTransformer/worker.js#L297), so that it re-interprets scopes correctly. I'll put up a fix today.

Thanks for the small repro, it'd have been impossible to fix the bug without it!

**Comment by janicduplessis** [comment]
@mjesun Awesome! Thanks for figuring out a fix, I'm not super familiar with babel :)

**Comment by janicduplessis** [comment]
@mjesun Thanks! I ended up just renaming the `r` variable since that lib was easy to fork.

Fixed in 59c8b850a2ec1b25f0f6a994ef6701a44284e951
